packages/grafana-ui/src/components/Logs/LogRow.tsx

```diff
@@ -67,49 +80,88 @@ class UnThemedLogRow extends PureComponent<Props, State> {
-          <div className={cx([style.logsRowDuplicates])}>
-            {row.duplicates && row.duplicates > 0 ? `${row.duplicates + 1}x` : null}
-          </div>
-        )}
-        <div className={cx([style.logsRowLevel])} />
-        {showTime && showUtc && (
-          <div className={cx([style.logsRowLocalTime])} title={`Local: ${row.timeLocal} (${row.timeFromNow})`}>
-            {row.timeUtc}
-          </div>
-        )}
-        {showTime && !showUtc && (
-          <div className={cx([style.logsRowLocalTime])} title={`${row.timeUtc} (${row.timeFromNow})`}>
-            {row.timeLocal}
-          </div>
-        )}
-        {showLabels && (
-          <div className={cx([style.logsRowLabels])}>
-            <LogLabels
-              getRows={getRows}
-              labels={row.uniqueLabels ? row.uniqueLabels : {}}
-              onClickLabel={onClickLabel}
-            />
+      <div>
+        <div
+          className={cx([style.logsRow])}
+          onClick={async () => {
+            this.setState({
+              showDetails: !showDetails,
+            });
+            if (this.state.linkModels === null) {
+              this.setState({
+                loadingDerivedFields: true,
+              });
+              const linkModels = await getDerivedFields(row.dataFrameRow);
+              if (this.mounted) {
+                this.setState({
+                  loadingDerivedFields: false,
+                  linkModels,
+                });
+              }
+            }
+          }}
+        >
+          {showDuplicates && (
+            <div className={cx([style.logsRowDuplicates])}>
+              {row.duplicates && row.duplicates > 0 ? `${row.duplicates + 1}x` : null}
+            </div>
+          )}
+          <div className={cx([style.logsRowLevel])} />
+          {showTime && showUtc && (
+            <div className={cx([style.logsRowLocalTime])} title={`Local: ${row.timeLocal} (${row.timeFromNow})`}>
+              {row.timeUtc}
+            </div>
+          )}
+          {showTime && !showUtc && (
+            <div className={cx([style.logsRowLocalTime])} title={`${row.timeUtc} (${row.timeFromNow})`}>
+              {row.timeLocal}
+            </div>
+          )}
+          {showLabels && (
+            <div className={cx([style.logsRowLabels])}>
+              <LogLabels
+                getRows={getRows}
+                labels={row.uniqueLabels ? row.uniqueLabels : {}}
+                onClickLabel={onClickLabel}
+              />
+            </div>
+          )}
+          <LogRowMessage
+            highlighterExpressions={highlighterExpressions}
+            row={row}
+            getRows={getRows}
+            errors={errors}
+            hasMoreContextRows={hasMoreContextRows}
+            updateLimit={updateLimit}
+            context={context}
+            showContext={showContext}
+            onToggleContext={this.toggleContext}
+          />
+        </div>
+        {showDetails && (
```

Comment:
Provisional way to show the DataLinks in LogRow but will be changed to the LogRow detail from https://github.com/grafana/grafana/pull/20034